Fix byte compilation of efc.el, from upstream SVN repository.
http://jdee.svn.sourceforge.net/viewvc/jdee/branches/2.3.6/jde/lisp/efc.el?r1=37&r2=43

--- jde/lisp/efc.el
+++ jde/lisp/efc.el
@@ -648,7 +648,7 @@
   (if (efc-coll-type-compatible-p this item)
       (oset this items (append (oref this items) (list item)))
     (error "Tried to add an item of type %s to a list of items of type %s"
-	   (typep item) (oref this elem-type))))
+	   (type-of item) (oref this elem-type))))
 
 (defmethod efc-coll-iterator ((this efc-list))
   "Return an iterator for this list."
@@ -760,14 +760,14 @@
 (defmethod initialize-instance ((this efc-hash-table) &rest fields)
   "Hash table constructor."
   (call-next-method)
-  (oset table (make-hash-table)))
+  (oset this table (make-hash-table)))
 
 (defmethod efc-coll-put ((this efc-hash-table) key value)
   "Put an item into the table."
   (if (efc-coll-type-compatible-p this value)
       (puthash key value (oref this table))      
     (error "Tried to add an item of type %s to a hash table of items of type %s"
-	   (typep value) (oref this elem-type))))
+	   (type-of value) (oref this elem-type))))
 
 (defmethod efc-coll-get ((this efc-hash-table) key)
   "Get an item from the table."
